Framework/MDAlgorithms/src/MDTransfModQ.cpp

```diff
@@ -166,22 +166,28 @@ the algorithm and false otherwise.
 * it also uses preprocessed detectors positions, which are calculated by
 PreprocessDetectors algorithm and set up by
 * calcYDepCoordinates(std::vector<coord_t> &Coord,size_t i) method.    */
-bool MDTransfModQ::calcMatrixCoordInelastic(const double &E_tr,
+bool MDTransfModQ::calcMatrixCoordInelastic(const double &deltaE,
                                             std::vector<coord_t> &Coord) const {
-  if (E_tr < m_DimMin[1] || E_tr >= m_DimMax[1])
+  if (deltaE < m_DimMin[1] || deltaE >= m_DimMax[1])
     return false;
-  Coord[1] = static_cast<coord_t>(E_tr);
-  double k_tr;
-  // get module of the wavevector for scattered neutrons
+  Coord[1] = static_cast<coord_t>(deltaE);
+
+  // x,y,z refer to internal coordinate system where Z is the beam direction
+  double qx{0.0}, qy{0.0}, qz{0.0};
   if (this->m_Emode == Kernel::DeltaEMode::Direct) {
-    k_tr = sqrt((m_Ei - E_tr) / PhysicalConstants::E_mev_toNeutronWavenumberSq);
+    const double kFinal = sqrt((m_eFixed - deltaE) /
+                               PhysicalConstants::E_mev_toNeutronWavenumberSq);
+    qx = -m_ex * kFinal;
+    qy = -m_ey * kFinal;
+    qz = m_kFixed - m_ez * kFinal;
   } else {
-    k_tr = sqrt((m_Ei + E_tr) / PhysicalConstants::E_mev_toNeutronWavenumberSq);
+    const double kInitial = sqrt(
+        (m_eFixed + deltaE) / PhysicalConstants::E_mev_toNeutronWavenumberSq);
+    qx = -m_ex * m_kFixed;
+    qy = -m_ey * m_kFixed;
+    qz = kInitial - m_ez * m_kFixed;
```

Comment:
this seems to be the minus of the original code, unless I have missed something

Reply:
This was @AndreiSavici 's point and we have an option in mantid already to set the convention to use `kf-ki` instead. Andrei pointed out that we can't change the default and we should already have this covered by enabling/disabling the `Crystallography Convention` option as appropriate in `View->Preferences->Mantid`.

Reply:
Can we try that?

Reply:
When I did the maths with @DavidFair yesterday, we got it to be with the minus sign. If the values of qz happen to be close to zero, there is a good chance the error would not be noticed.

Reply:
It should depend on the convention: Inelastic usually uses `ki-kf` where crystallography uses `kf-ki` or do you just mean the `qz` component and not all of them?

Reply:
I cant remember which convention we used, my notes are at home. I can check on Monday.